models/sessions/scratch/bigquery/snowplow_web_sessions_this_run.sql

```diff
@@ -166,6 +199,41 @@ session_aggs as (
         {% endif %}
     group by
         domain_sessionid
+),
+
+event_counts as (
+    select
+        domain_sessionid,
+        array_agg(struct(
+            name,
+            counts
+        )) as session_events,
```

Comment:
I think it would be good to make this optional with a variable, as it's quite an intense operation that may not be relevant to most users. 

I'm also not entirely sure how someone would _use_ this field, for example how would I filter the data to just include sessions that had a `link_click` event? I don't have a better structure to make this easier but is that kind of query currently possible?

Reply:
Yeh you're right, unfortunately BQ has no object data type which will make it easy to query for specific count of events per session (a row level field); it was included as it is a metric that is filterable for the GA4 engagement overview report.

I think we'll remove this field from this PR and revisit this again.

Reply:
Makes sense, annoyingly we can do it with PIVOT but we'd have to know the list of event_names in advance or do it using dynamic SQL which I don't know if plays nice with dbt (unlikely...). Raise an issue for it so we don't forget, but makes sense to remove from this pr!